chain/index/chain-indexing-overview-for-rpc-providers.MD

```diff
@@ -0,0 +1,218 @@
+      },
+      {
+        "/": "bafy2bzacedgei4ve3spkfp3oou5oajwd5cogn7lljsuvoj644fgj3gv7luamu"
+      },
+      {
+        "/": "bafy2bzacebbpcnjoi46obpaheylyxfy5y2lrtdsyglqw3hx2qg64quip5u76s"
+      }
+    ],
+    "Height": 1995103,
+    "IndexedMessagesCount": 0,
+    "IndexedEventsCount": 0,
+    "Backfilled": true,
+    "IsNullRound": false
+  }
+}
+```
+
+### `lotus-shed chainindex validate-backfill` CLI tool
+The `lotus-shed chainindex validate-backfill` command is a tool for validating and optionally backfilling the chain index over a range of epochs since calling the `ChainValidateIndex` API for a single epoch at a time can be cumbersome, especially when backfilling or validating the index over a range of historical epochs, such as during a migration. It wraps the `ChainValidateIndex` API to efficiently process multiple epochs.
+
+**Note: This command can only be run when the Lotus daemon is already running with the [`ChainIndexer` enabled](#enablement) as it depends on the `ChainValidateIndex` RPC API.**
+
+#### Usage:
+```
+lotus-shed chainindex validate-backfill --from <start_epoch> --to <end_epoch> [--backfill] [--log-good]
+```
+
+The command validates the chain index entries for each epoch in the specified range, checking for missing or inconsistent entries(i.e. the indexed data does not match the actual chain state). If `--backfill` is enabled (which it is by default), it will attempt to backfill any missing entries using the `ChainValidateIndex` API.
+
+You can learn about how to use the tool with `lotus-shed chainindex validate-backfill -h`.
+
+## Downgrade
+
+In case you need to downgrade to the [previous indexing system](#previous-indexing-system), follow these steps:
+
+1. Stop your Lotus node.
+2. Build your Lotus binary for the rollback version which has the implementation of the old `EthTxHashLookup`, `MsgIndex`, and `EventIndex` indices.
+4. Ensure that you've set the correct config for the existing `EthTxHashLookup`, `MsgIndex`, and `EventIndex` indices in the `config.toml` file.
+5. Restart your Lotus node.
+6. Backfill the `EthTxHashLookup`, `MsgIndex`, and `EventIndex` indices using the `lotus-shed index backfill-*` CLI tooling available in the [previous indexing system](#previous-indexing-system) for the epoch range in [epochs between the upgrade to `ChainIndexer` and the rollback of `ChainIndexer`].
+
+## Terminology
+### Previous Indexing System
+* This corresponds to the indexing system used in Lotus versions before v1.31.0.  
+* It has been replaced by the [ChainIndex](#chainindexer-indexing-system).
+* It was composed of three indexers using three separate databases: [`EthTxHashLookup`](https://github.com/filecoin-project/lotus/blob/master/chain/ethhashlookup/eth_transaction_hash_lookup.go), [`MsgIndex`](https://github.com/filecoin-project/lotus/blob/master/chain/index/msgindex.go), and [`EventIndex`](https://github.com/filecoin-project/lotus/blob/master/chain/events/filter/index.go).
```

Comment:
I think these links are going to go away when you merge, so you could just leave them off I think, code links for users isn't usually very useful

Reply:
This was me.  I was thinking to update them to v1.30.0 tag.  I had them as a way to give a pointer for someone to look at what was actually happening.  I was ok with it because it didn't obscure things for the reader of the rendered markdown, while giving a self-service way for someone to learn more if they wanted to.

Reply:
ok, tagging wouldn't be terrible I guess

I still question the utility of pointing people at code, but it's not a blocker if you think it preferable

Reply:
Handling in https://github.com/filecoin-project/lotus/pull/12600